File: notebooks/scala/common/DatetimeUtils.scala
// Databricks notebook source
// Retrieve timestamp and datestamp upon notebook startup
import java.util.Date
import java.time.format.DateTimeFormatter
import java.text.SimpleDateFormat

class CurrentTime {
  val sdf = new SimpleDateFormat("yyyyMMdd")
  val rdf = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss:SSS");
  val currentTime = System.currentTimeMillis

  def getTimestamp() = currentTime/1000
  
  def getDatestamp() = sdf.format(new Date(currentTime))
  
  def getRedshiftTimestamp() = rdf.format(new Date(currentTime))
}

File: notebooks/python/common/datetime_utils.py
# Databricks notebook source
from datetime import datetime

# COMMAND ----------

class Date:
    
    def __init__(self):
        self.date = datetime.today()
        
    def getDatestamp(self, output_string_format: str = "%Y%m%d"):
        datestamp = self.date.strftime(output_string_format)
        return datestamp

    def getTimestamp(self):
        timestamp = str(int(self.date.timestamp()))
        return timestamp
